examples/nemo/eg6/dummy.f90

```diff
@@ -0,0 +1,35 @@
+program dummy
+   real, dimension(10, 10)   :: umask
+   character(len=10) :: char_var
+   integer  :: ji, jj
+   logical :: logical_var
+   integer(kind=8) :: offset
+   integer, intrinsic :: loc, sizeof
+
+   logical_var = .false.
+   char_var = "test"
+
+   do jj = 1, 10
+      do ji = 1, 10
+         if (char_var .eq. "abc" .and. logical_var) then
+            umask(ji,jj) = 1
+         else
+            umask(ji,jj) = 3
+         endif
+       end do
+
+    ! VERY NAUGHTY CODE AHEAD - there be dragon!!
+    ! We modify the values of some read-only fields by using offsets
+    ! in a written array. This will abort if the code is compiled
+    ! with array bound check of course.
+    ! TODO: Unfortunately, PSyclone does not support loc or sizeof,
```

Comment:
Please could you clarify here. I tried it for myself and see that it's the generation of the PSyData calls that is broken. If we don't already have an Issue please could you create one and refer to it here. Generating code that doesn't compile isn't a good look for us :-(

Reply:
Not really a PSyclone problem (at this stage at least :) ). The two functions `loc` and `sizeof` are non-standard extensions, and as such are not detected by fparser (and then end up as being assumed to be arrays).  We have an fparser issue to support additional (standard) intrinsics:
https://github.com/stfc/fparser/issues/434

So I've added these two in with the comment that they are non-standard. I've also updated the comment to better indicate why it doesn't work.

If you prefer, I could handle `loc` and `sizeof` as special case in the transformation, so that they wouldn't get 'extracted' ... but of course that would mean if a user has those names used as variables, it would fail as well :(

Reply:
No, that's good now, thanks.